tests/Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests/ProjectSystem/VS/Properties/VBDefineConstantsEncodingTests.cs

```diff
@@ -0,0 +1,135 @@
+        [InlineData("a b c d=/\"value1/\"", new[] { "a b c d", "\"value1\"" })]
+        [InlineData("a=/\"b/=c/=d/=value1/\"", new[] { "a", "\"b=c=d=value1\"" })]
+        [InlineData("a=/\"b/\"/\"c/=d/=value1/\"", new[] { "a", "\"b\"\"c=d=value1\"" })]
+        [InlineData("/\"ab/\"/\"cd=/\"value1/\"", new[] { "\"ab\"\"cd", "\"value1\"" })]
+        [InlineData("/\"/\"ab/\"/\"cd/\"/\"ef=/\"value1/\"", new[] { "\"\"ab\"\"cd\"\"ef", "\"value1\"" })]
+        [InlineData("/\"/\"ab/\"/\"cd=/\"value1/\"", new[] { "\"\"ab\"\"cd", "\"value1\"" })]
+        [InlineData("a//bc=/\"value1/\"", new[] { "a/bc", "\"value1\"" })]
+        [InlineData("", new string[0])]
+
+        [InlineData("a=1,b=2", new[] { "a", "1", "b", "2" })]
+        [InlineData("a=1123,b=456", new[] { "a", "1123", "b", "456" })]
+        [InlineData("a=1,b=/\"apple/\"", new[] { "a", "1", "b", "\"apple\"" })]
+        [InlineData("a=1,b=/\"apple/\",c=456", new[] { "a", "1", "b", "\"apple\"", "c", "456" })]
+        [InlineData("a=1,b=/\"apple/\",c=456,d=True", new[] { "a", "1", "b", "\"apple\"", "c", "456", "d", "True" })]
+
+        [InlineData("a=True", new[] { "a", "True"})]
+        [InlineData("a=False", new[] { "a", "False" })]
+        [InlineData("a=True,b=False", new[] { "a", "True", "b", "False" })]
+        [InlineData("a=True,b=False,c=123", new[] { "a", "True", "b", "False", "c", "123" })]
+        public void NameValuePairsVBDefineConstantsEncoding(string encodedPairs, string[] pairs)
+        {
+            VBDefineConstantsEncoding _encoding = new();
+
+            Assert.Equal(expected: pairs, actual: _encoding.Parse(encodedPairs).SelectMany(pair => new[] { pair.Name, pair.Value }));
+
+            Assert.Equal(encodedPairs, _encoding.Format(ToNameValues(pairs)));
+
+            static IEnumerable<(string Name, string Value)> ToNameValues(IEnumerable<string> pairs)
+            {
+                using var e = pairs.GetEnumerator();
+                while (e.MoveNext())
+                {
+                    var name = e.Current;
+                    Assert.True(e.MoveNext());
+                    var value = e.Current;
+                    yield return (name, value);
+                }
+            }
+        }
+
+        [Theory]
+        [InlineData("=key1=")]
+        [InlineData("=key1")]
+        [InlineData(",key1")]
+        [InlineData(",,,key1")]
+        [InlineData("key1,")]
+        [InlineData("key1=value1,")]
+        [InlineData("key1,,,")]
+        [InlineData("=")]
+        [InlineData("==")]
+        [InlineData("===")]
+        [InlineData("key1,=abcd")]
+        [InlineData("key1,=,abcd")]
+        [InlineData("=\"\"")]
+        [InlineData(",")]
+        [InlineData(",,,")]
+        public void InvalidVBDefineConstantsEncoding(string input)
+        {
+            VBDefineConstantsEncoding _encoding = new();
+            FormatException exception = Assert.Throws<FormatException>(() => (_encoding.Format(_encoding.Parse(input))));
```

Comment:
Is it the `Format` or the `Parse` that you expect to throw here? Narrowing the code within the `Assert.Throws` makes the test more targeted.

My intuition is that you only need the `Parse` here anyway.